### ISSUE TITLE
[BUG] Azure storage files ShareClient throwing an exception

### ISSUE THREAD

**Issue by SridharPasham** [Azure/azure-sdk-for-net#15828]
**Describe the bug**
Getting error when called CreateIfNotExists() on ShareClient instance. Error: "Retry failed after 6 tries" with InnerException message "The underlying connection was closed: An unexpected error occurred on a send."

**Expected behavior**
The error should provide some useful information. This error is not useful at all.

**Actual behavior (include Exception or Stack Trace)**
Throwing Aggregate Exception.

` at Azure.Core.Pipeline.RetryPolicy.<ProcessAsync>d__11.MoveNext()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Azure.Core.Pipeline.RetryPolicy.Process(HttpMessage message, ReadOnlyMemory`1 pipeline)
   at Azure.Core.Pipeline.HttpPipelinePolicy.ProcessNext(HttpMessage message, ReadOnlyMemory`1 pipeline)
   at Azure.Core.Pipeline.HttpPipelineSynchronousPolicy.Process(HttpMessage message, ReadOnlyMemory`1 pipeline)
   at Azure.Core.Pipeline.HttpPipelinePolicy.ProcessNext(HttpMessage message, ReadOnlyMemory`1 pipeline)
   at Azure.Core.Pipeline.HttpPipelineSynchronousPolicy.Process(HttpMessage message, ReadOnlyMemory`1 pipeline)
   at Azure.Core.Pipeline.HttpPipelinePolicy.ProcessNext(HttpMessage message, ReadOnlyMemory`1 pipeline)
   at Azure.Core.Pipeline.HttpPipelineSynchronousPolicy.Process(HttpMessage message, ReadOnlyMemory`1 pipeline)
   at Azure.Core.Pipeline.HttpPipeline.Send(HttpMessage message, CancellationToken cancellationToken)
   at Azure.Storage.Files.Shares.FileRestClient.Share.<CreateAsync>d__0.MoveNext()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Azure.Storage.Files.Shares.ShareClient.<CreateInternal>d__31.MoveNext()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Azure.Storage.Files.Shares.ShareClient.<CreateIfNotExistsInternal>d__34.MoveNext()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Azure.Storage.Files.Shares.ShareClient.CreateIfNotExists(IDictionary`2 metadata, Nullable`1 quotaInGB, CancellationToken cancellationToken)
   at FileStorageTest.Program.Main(String[] args) in C:\Tests\FileStorageTest\FileStorageTest\Program.cs:line 24`

**To Reproduce**
Created a sample console application and I am trying to connect to Azure file storage using this code.
`
var connectionString = "DefaultEndpointsProtocol=https;AccountName=account-name;AccountKey=account-key;EndpointSuffix=core.windows.net;";
var shareClient = new ShareClient(connectionString, "testfileshare");
shareClient.CreateIfNotExists();`

**Environment:**
 - Azure.Storage.Files.Shares, 12.4.0 version 
 - Windows 10, .NET4.6.2 also tried .NET4.7.2
 - VisualStudio 2019, 16.7

**Note**
When I use Microsoft.Azure.Storage.File, 11.1.3 and using this code works fine for accessing files.
`var storageAccount = CloudStorageAccount.Parse("FileStorageConnectionString");
    var fileClient = storageAccount.CreateCloudFileClient();
    var share = fileClient.GetShareReference("StorageFileShareName");
    var file = this.share.GetRootDirectoryReference().GetFileReference("file.txt");`

I need lease functionality but unfortunately only 12.0.0 SDK providing this (11.1.3 also supports using REST API but I want SDK way)

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team best able to assist.

**Comment by SridharPasham** [comment]
It is working fine now. Don't know whether its caused because of VS or Nuget feed. 
Deleted the sample and closed all VS instances and recreated the application and it worked fine.

**Comment by SridharPasham** [comment]
I have found the issue. When I add Azure.Storage.Files.Shares package reference from nuget I observed that in project file Azure.Core reference is adding from "net461" folder and others are adding from "netstandard2.0" folder and my application target framework is "Net4.6.2"

This causing the issue. When I manually changed Azure.Core reference path to "netstandard2.0" it is working fine. 

        `<Reference Include="Azure.Core, Version=1.5.0.0, Culture=neutral, PublicKeyToken=92742159e12e44c8, processorArchitecture=MSIL">
    
          <HintPath>..\packages\Azure.Core.1.5.0\lib\netstandard2.0\Azure.Core.dll</HintPath>
        </Reference>`

I have reproduced this issue by upgrading "Azure.Storage.Files.Shares" package. I do not know whether it is VS issue or nuget issue.
